pkg/runtime/helper.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
 )
 
+// TODO: move me to pkg/api/meta
```

Comment:
Originally pkg/runtime was an opinionated part of conversion.  Things strictly related to decoding generic objects in JSON showed up here.  I think where we are today is that we want to have the core "smart object" model (pkg/smartobjects) which can do defaulting, conversion, new, etc, and then a set of serialization choices (JSON, query, protobufs, msgpack, whatever) laid on top of that (orthogonal to).  The things in pkg/runtime that relate to the "scheme" (conversion, defaults) should move to smart objects.  The things related to JSON serialization should be split out.  I'm still torn on what we should do w.r.t. interfaces for helpful methods.  

Ideally, I would think the smart objects code would have the ability to generate boilerplate helper methods (like ObjectMeta() or SetObjectMeta(...)) to your scheme.  That would reduce the amount of code in api/meta (which is really about just having a common interface on all of our api objects to get and set metadata).  We had just balked at adding one getter per type as being painful and hadn't got further.